pkg/skaffold/build/dependencies.go

```diff
@@ -83,18 +83,18 @@ func DependenciesForArtifact(ctx context.Context, a *latest.Artifact, cfg docker
 }
 
 // CreateBuildArgsFromArtifacts creates docker build args for an artifact from its required artifacts slice.
-func CreateBuildArgsFromArtifacts(deps []*latest.ArtifactDependency, r ArtifactResolver) (map[string]*string, error) {
+func CreateBuildArgsFromArtifacts(deps []*latest.ArtifactDependency, r ArtifactResolver) map[string]*string {
 	if r == nil {
 		// `diagnose` is called without an artifact resolver. Return an empty map in this case.
-		return nil, nil
+		return nil
 	}
 	m := make(map[string]*string)
 	for _, d := range deps {
-		t, err := r.GetImageTag(d.ImageName)
-		if err != nil {
-			return nil, err
+		t, found := r.GetImageTag(d.ImageName)
+		if !found {
+			return nil
```

Comment:
Would it make sense to have the map have nil pointers for not-found cases?
```suggestion
		if t, found := r.GetImageTag(d.ImageName); found {
		         m[d.Alias] = &t
		} else {		
			m[d.Alias] = nil
		}
```